internal/provider-simple-v6/state_store_inmem.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package simple
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform/internal/backend"
+	"github.com/hashicorp/terraform/internal/configs/configschema"
+	"github.com/hashicorp/terraform/internal/providers"
+	"github.com/hashicorp/terraform/internal/tfdiags"
+	"github.com/zclconf/go-cty/cty"
+)
+
+const inMemStoreName = "simple6_inmem"
+
+// InMemStoreSingle allows 'storing' state in memory for the purpose of testing.
+//
+// "Single" reflects the fact that this implementation does not use any global scope vars
+// in its implementation, unlike the current inmem backend. HOWEVER, you can test whether locking
+// blocks multiple clients trying to access the same state at once by creating multiple instances
+// of backend.Backend that wrap the same provider.Interface instance.
+type InMemStoreSingle struct {
+	states stateMap
+	locks  lockMap
+}
+
+func stateStoreInMemGetSchema() providers.Schema {
+	return providers.Schema{
+		Body: &configschema.Block{
+			Attributes: map[string]*configschema.Attribute{
+				"lock_id": {
+					Type:        cty.String,
+					Optional:    true,
+					Description: "initializes the state in a locked configuration",
+				},
+			},
+		},
+	}
+}
+
+func (m *InMemStoreSingle) ValidateStateStoreConfig(req providers.ValidateStateStoreConfigRequest) providers.ValidateStateStoreConfigResponse {
+	// Not implemented in original inmem backend.
+	// The inmem used default logic in the backendbase package that cannot be replicated here easily.
+	// Instead, here is a rough implementation of validation:
+	var resp providers.ValidateStateStoreConfigResponse
+
+	attrs := req.Config.AsValueMap()
+	if v, ok := attrs["lock_id"]; ok {
+		if !v.IsKnown() {
```

Comment:
I can maybe imagine why this could be null but under what circumstances would it be unknown? I thought the entire concept of "known/unknown" (known "later") is relevant during planning but it does not apply (pun not intended) to backends/PSS, or does it?